src/pages/Viewer/Viewer.js

```diff
@@ -1,7 +1,338 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { Document, Page, pdfjs } from 'react-pdf';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import Advances_in_Coumputer_Science from '../../../src/components/PDF/Advances_in_Computer_Science.pdf';
+import styled from 'styled-components';
+import 'react-pdf/dist/esm/Page/TextLayer.css';
+import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
+
+pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;
 
 const Viewer = () => {
-  return <div>This is Viewer~!!!!!</div>;
+  const [bookInfo, setBookInfo] = useState([]);
+  const [numPages, setNumPages] = useState(null);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [pageScale, setPageScale] = useState(1);
+  const { id, title, list, list_number, content_url } = bookInfo;
+  const params = useParams();
+  const bookid = params.bookId;
```

Comment:
구조분해할당해서 사용할 수도 있습니다!
```suggestion
  const { bookId } = useParams();
```